src/virtual.jl

```diff
@@ -0,0 +1,36 @@
+# virtual dataset
+struct VirtualMapping
+  vspace::Dataspace
+  srcfile::String
+  srcdset::String
```

Comment:
Could we provide a convenience constructor that accepts `AbstractString` and converts them to `String`?

Reply:
I'm pretty sure this already works.